docs/internals/optimizer.rst

```diff
@@ -138,4 +184,669 @@ So the call to function ``f`` is inlined and the original definition of ``f`` ca
 Inlining like this is attempted, whenever a heuristics suggests that inlining is cheaper over the lifetime of a
 contract than not inlining. This heuristics depends on the size of the function body, the
 number of other references to its tag (approximating the number of calls to the function) and
-the expected number of executions of the contract (the global optimiser parameter "runs").
+the expected number of executions of the contract (the global optimizer parameter "runs").
+
+Yul-Based Optimizer Module
+==========================
+
+The Yul-based optimizer consists of several stages and components that all transform
+the AST in a semantically equivalent way. The goal is to end up either with code
+that is shorter or at least only marginally longer but will allow further
+optimization steps.
+
+The optimizer currently follows a purely greedy strategy and does not do any
+backtracking.
+
+All components of the Yul-based optimizer module are explained below.
+The following transformation steps are the main components:
+
+ - SSA Transform
+ - Common Subexpression Eliminator
+ - Expression Simplifier
+ - Redundant Assign Eliminator
+ - Full Function Inliner
+
+Selecting optimizations
+~~~~~~~~~~~~~~~~~~~~~~~
+
+By default the optimizer applies its predefined sequence of optimization steps to
+the generated assembly. You can override this sequence and supply your own using
+the `--yul-optimizations` option:
+
+``` bash
+solc --optimize --ir-optimized --yul-optimizations 'dhfoD[xarrscLMcCTU]uljmul'
+```
+
+Available abbreviations are listed in the `Yul optimizer docs </docs/yul.rst#optimization-step-sequence>`_.
+
+Preprocessing
+~~~~~~~~~~~~~
+
+The preprocessing components perform transformations to get the program
+into a certain normal form that is easier to work with. This normal
+form is kept during the rest of the optimization process.
+
+Disambiguator
+~~~~~~~~~~~~~
+
+The disambiguator takes an AST and returns a fresh copy where all identifiers have
+names unique to the input AST. This is a prerequisite for all other optimizer stages.
+One of the benefits is that identifier lookup does not need to take scopes into account
+and we can basically ignore the result of the analysis phase.
```

Comment:
What analysis phase are we referring to here?

Reply:
I don't know, I took this text as if from the Yul-optimizer readme.